libraries/plugins/xfpga/sysfs.c

```diff
@@ -2451,8 +2454,8 @@ fpga_result make_sysfs_object(char *sysfspath, const char *name,
 	}
 	obj->handle = handle;
 	obj->type = FPGA_SYSFS_FILE;
-	obj->buffer = calloc(objstat.st_size, sizeof(uint8_t));
-	obj->max_size = objstat.st_size;
+	obj->max_size = MIN_SYSOBJECT_FILESIZE;
```

Comment:
what if the object.st_size is bigger than MIN_SYSOBJECT_FILESIZE?

Reply:
That was the nature of the bug.  The objstat.st_size for a file containing "0" was 2.  But, the subsequent write size was for a string that contained "0x400000000" (which is greater than 2, and so failed the length check).  Should it be set to max(objstat.st_size, MIN_SYSOBJECT_FILESIZE)?

Reply:
That's what I was expecting. max of the file size as reported by stat or the minimum.